### ISSUE TITLE
This is not hexagonal architecture

### ISSUE THREAD

**Issue by MyEidos** [Amitpnk/Hexagonal-architecture-ASP.NET-Core#2]
The current project doesn't satisfy the requirements of hexagonal architecture. 
Ex: concrete (!) EF DB context (!) is used on the domain layer:
public DealDomain(ApplicationDbContext dbContext)

Which creates strong coupling with DB layer.

**Comment by Innokmm** [comment]
I agree, the first thing I've noticed in the Domain layer is a reference to Persistence, which as per comment from MyEidos ends up as dependency to the EF:

![image](https://user-images.githubusercontent.com/29246399/145276560-1d012159-450d-4a89-87d9-adbde2c0ba6c.png)

Shouldn't Persistence depend on Domain layer?
